docs/frontend/Modern-JS-features.md

```diff
@@ -0,0 +1,498 @@
+# Modern Javascript Features                                            <!-- omit in toc -->
+## Allowing frontend development using up to date Javascript features   <!-- omit in toc -->
+
+- [Introduction](#Introduction)
+- [Tooling](#Tooling)
+  - [Babel](#Babel)
+- [ECMAScript modern features](#ECMAScript-modern-features)
+  - [Arrow functions](#Arrow-functions)
+    - [Passing functions as predicates](#Passing-functions-as-predicates)
+    - [Differences between arrow functions and typical functions](#Differences-between-arrow-functions-and-typical-functions)
+  - [String interpolation](#String-interpolation)
+  - [Classes](#Classes)
+  - [`async` and `await`](#async-and-await)
+    - [Basic asynchronous functions](#Basic-asynchronous-functions)
+    - [Blocking vs non-blocking functions](#Blocking-vs-non-blocking-functions)
+    - [Using `async` and `await` to write more concise code](#Using-async-and-await-to-write-more-concise-code)
+      - [Asynchronicity libraries (Like async.js)](#Asynchronicity-libraries-Like-asyncjs)
+      - [Promises](#Promises)
+      - [`async` and `await` syntactic sugar](#async-and-await-syntactic-sugar)
+  - [Modules](#Modules)
+    - [CommonJS](#CommonJS)
+    - [AMD - Asynchronous Module Definition](#AMD---Asynchronous-Module-Definition)
+    - [ECMAScript modules](#ECMAScript-modules)
```

Comment:
Nice use of ToC 👏

Reply:
Always.  I have a plugin for generating them.